### PR TITLE
[pigeon] Swift implementation for ProxyApis

### PR DESCRIPTION
**Pull request by bparrishMines**
Swift portion of https://github.com/flutter/flutter/issues/134777

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Review by stuartmorgan — APPROVED** [review]
LGTM, but please wait for Tarrin's approval on the generator changes.

I did a relatively light review pass here, but the the instance manager looks like I expect from the Obj-C webview_flutter prototype, and the overall structure and the availability annotations make sense to me.

File: packages/pigeon/lib/swift_generator.dart
Comment (tarrinneal) [inline packages/pigeon/lib/swift_generator.dart:2450]: I think it does more than this? Either way, needs a period.

**Review by tarrinneal — APPROVED** [review]
Not seeing any real problems here, seems like all of the issues from kotlin have been fixed here already. If the code works, I think it's good to go.

File: packages/pigeon/platform_tests/test_plugin/example/ios/RunnerTests/InstanceManagerTests.swift
Comment (hellohuanlin) [inline packages/pigeon/platform_tests/test_plugin/example/ios/RunnerTests/InstanceManagerTests.swift:28]: optional nit: can use `XCTUnwrap` to avoid `!`. When it's nil, instead of crashing, it would mark it as a test failure and continue with other tests.

Comment (bparrishMines) [reply packages/pigeon/platform_tests/test_plugin/example/ios/RunnerTests/InstanceManagerTests.swift]: I changed the following line to: 

```swift
XCTAssertEqual(instanceManager.instance(forIdentifier: try XCTUnwrap(identifier)), object)
```

Is that what you were thinking? The `XCTAssertNotNil(identifier)` check also fails the test but continues to run other tests.

**Pull request merged**
